docs/user.adoc

```diff
@@ -1142,6 +1142,90 @@ c5e80c99-d7c3-4acb-a5e1-73592ceebbbc fred/******
+
+The CLI Command for listing and create credentials and domains in XML format in a credentials store is `list-credentials-as-xml`.
+
+.Example of listing the credentials and domains in the `/example-folder` folder.
+[source,bash]
+----
+$ java -jar jenkins-cli.jar -s https://jenkins.example.com/ \
+list-credentials-as-xml folder::item::/example-folder
+<list>
+  <com.cloudbees.plugins.credentials.domains.DomainCredentials plugin="credentials@2.1.17-SNAPSHOT">
+    <domain>
+      <specifications/>
+    </domain>
+    <credentials>
+      <com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl>
+        <id>7d9702ee-9ce3-4ecd-82b5-9ebe125459fb</id>
+        <description></description>
+        <username>bob</username>
+        <password>
+          <secret-redacted/>
+        </password>
+      </com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl>
+    </credentials>
+  </com.cloudbees.plugins.credentials.domains.DomainCredentials>
+  <com.cloudbees.plugins.credentials.domains.DomainCredentials plugin="credentials@2.1.17-SNAPSHOT">
+    <domain>
+      <name>production</name>
+      <description>Credentials used for the production roadrunner cooking infrastructure</description>
+      <specifications/>
+    </domain>
+    <credentials>
+      <com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl>
+        <id>ae68086e-ccb1-43cc-8820-4f111bceda7c</id>
+        <description></description>
+        <username>wilma</username>
+        <password>
+          <secret-redacted/>
+        </password>
+      </com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl>
+    </credentials>
+  </com.cloudbees.plugins.credentials.domains.DomainCredentials>
+  <com.cloudbees.plugins.credentials.domains.DomainCredentials plugin="credentials@2.1.17-SNAPSHOT">
+    <domain>
+      <name>testing</name>
+      <description>Credentials for use during testing of any new recipes</description>
+      <specifications/>
+    </domain>
+    <credentials>
+      <com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl>
+        <id>deploy-key</id>
+        <description></description>
+        <username>wecoyote</username>
+        <password>
+          <secret-redacted/>
+        </password>
+      </com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl>
+      <com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl>
+        <id>c5e80c99-d7c3-4acb-a5e1-73592ceebbbc</id>
+        <description></description>
+        <username>fred</username>
+        <password>
+          <secret-redacted/>
+        </password>
+      </com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl>
+    </credentials>
+  </com.cloudbees.plugins.credentials.domains.DomainCredentials>
+</list>
+----
+
+The output of the command can be used as input (for --import) as is, the only needed change is to set the actual
+Secrets which are redacted in the output.
```

Comment:
How would a user have any idea that this “list” command can mutate?

How can a Jenkins admin lock down mutation while allowing read (eg if the Jenkins admin uses an init groovy script to filter out mutation CLI commands from the CLI command extension list.

I think import should be a separate command added to this PR

Reply:
> How would a user have any idea that this “list” command can mutate?

```
$> java -jar jenkins-cli.jar -noKeyAuth -s http://localhost:8080/jenkins/ help list-credentials-as-xml
java -jar jenkins-cli.jar list-credentials-as-xml STORE [--import]
Import and Export credentials as XML. The output of the command can be used as input (for --import) as is, the only needed change is to set the actual Secrets which are redacted in the output.
 STORE    : Store ID. For example the Jenkins credentials global store would be
            identified by "system::system::jenkins", and folder scoped
            credentials as "folder::item::/full/name/of/folder"
 --import : Read credentials as XML from the standard input and create them in
            the given store.
```

> How can a Jenkins admin lock down mutation while allowing read

Using permissions?

> if the Jenkins admin uses an init groovy script to filter out mutation CLI commands from the CLI command extension list.

Never saw such an exotic way to do that, but ok.

> I think import should be a separate command added to this PR

Ok, if you feel strongly... but I don't see the point.